src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs

```diff
@@ -50,6 +50,7 @@ protected static HttpClientHandler CreateHttpClientHandler(Version useVersion =
             {
                 SocketsHttpHandler socketsHttpHandler = (SocketsHttpHandler)GetUnderlyingSocketsHttpHandler(handler);
                 SetQuicImplementationProvider(socketsHttpHandler, quicImplementationProvider);
+                socketsHttpHandler.SslOptions.LocalCertificateSelectionCallback = null;
```

Comment:
This defaults to null.  What's setting it to non-null?

Reply:
AFAIU: `HttpClientHandler.ClientCertificateOptions `

Reply:
And it gets set in default ctor of client handler:
https://github.com/dotnet/runtime/blob/b823f14ab759b4cc32c7315b1c278c226dc2af20/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L48-L48

Reply:
I see, thanks.  What happens when this isn't nulled out?  If our QUIC implementation doesn't support a non-null LocalCertificateSelectionCallback, doesn't this mean we're making our tests pass even though all normal usage will fail?

Reply:
Before #55877 LocalCertificateSelectionCallback was just ignored silently, so I guess if something weird was passed in the callback we didn't catch it, if that's what you mean?

Reply:
No, I mean with:
https://github.com/dotnet/runtime/pull/55877/files#diff-28afe02b30ccd883819eb268198bb1a9adf9fb0ca74822dda6a5ead83c8b72e5R52-R55
anyone using `new HttpClient()` now will fail with HTTP/3, because HttpClientHandler's ctor is setting the property.  So we're basically making our tests workaround that and hide the fact that we've made the entire feature unusable ;-)

I think we should back out 55877 instead until a proper fix can be put in place.

Reply:
Just to be safe, I created PR with revert: #56097 ... we can decide which one to go with based on test results and based on discussion here.

Reply:
> So we're basically making our tests workaround that and hide the fact that we've made the entire feature unusable ;-)

That's true :( Mana mentioned in Teams that this "is rather a hack and not a fix" to unblock CI and that we would need to follow up with a proper fix obviously

Reply:
Thanks @karelz !

Reply:
I personally think that reverting might be safer because it's possible there were other things like this implicit LocalCertificateSelectionCallback... 🤔

Reply:
Looking into the code, all this hidden `HttpClientHandler` `LocalCertificateSelectionCallback` does is selects a cert from `ClientCertificates` collection based on having private key and checking its extensions. And that first part is what we do in msquic (selecting the one with private key). So despite ignoring the callback, it might still work as excepted in most cases. But @wfurt would have to chime in here for more details, this the extent of my knowledge with certs.

Reply:
The `HttpClientHandler` callback:
https://github.com/dotnet/runtime/blob/913facdca8b04cc674163e31a7650ef6868a7d5b/src/libraries/Common/src/System/Net/Security/CertificateHelper.cs#L29-L62

S.N.Quic code:
https://github.com/dotnet/runtime/blob/913facdca8b04cc674163e31a7650ef6868a7d5b/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs#L57-L72

It's not the same thing, but there are some some similarities...

Reply:
But @stephentoub is right, `new HttpClient` + H/3 request will crash anyway. My hack mitigates this only in out tests.

Reply:
reverting #56097 was the right choice. Sorry for the mess, I'm not sure how I missed it. The problem with `LocalCertificateSelectionCallback` is that it is not supported by MsQuic. so We either need to go back to ignoring it or fix the tests.